### ISSUE TITLE
Error when using namedtuple in parallel loop

### ISSUE THREAD

**Issue by HPLegion** [numba/numba#4846]
Hi numba team,

I encountered a bug today when I tried to call a compiled function from a parallel loop, I think this happens because I try to pass a namedtuple to the inner function.

An example:
```python
from numba import njit, prange
from collections import namedtuple

mytype = namedtuple("mytype", ("a", "b"))

@njit(parallel=True, nogil=True)
def outer(mydata):
    for k in prange(3):
        inner(k, mydata)

@njit(nogil=True)
def inner(k, mydata):
    print(k, mydata.a)
    print(k, mydata.b)

mydata = mytype(a="a", b="b")

# inner(100, mydata) ## -> Works as expected

outer(mydata) ## -> does not work
```
This produces the following error message:

>  numba.errors.LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
> scalar type mytype(unicode_type x 2) given for non scalar argument #2
> 
> File "temp/prangenamedtuple.py", line 8:
> def outer(mydata):
>     for k in prange(3):
>     ^
> 
> [1] During: lowering "id=0[LoopNest(index_variable = parfor_index.0, range = (0, $const2.2, 1))]{12: <ir.Block at temp/prangenamedtuple.py (8)>}Var(parfor_index.0, temp/prangenamedtuple.py (8))" at temp/prangenamedtuple.py (8)

This error occurs with either _nogil_ setting (True/False) and it disappears if I use the normal _range_ instead of _prange_ command (this obviously results in a warning that numba did not manage to parrallelise anything)

I am using numba version 0.46.0 from Anaconda
Please let me know if you need any more information from me :)

Thanks for your work!

**Comment by stuartarchibald** [comment]
Thanks for the report. I think this is a limitation of the current prange/gufunc implementation. It's because gufunc (the thing prange gets transformed to) wants arrays not tuples. This has the same problem:
```python
from numba import njit, prange

@njit(parallel=True, nogil=True)
def outer(mydata):
    for k in prange(3):
        inner(k, mydata)

@njit(nogil=True)
def inner(k, mydata):
    print(k, mydata[0])
    print(k, mydata[1])

mydata = (1, 2)
outer(mydata)
```

**Comment by stuartarchibald** [comment]
Closing as duplicate of meta-issue https://github.com/numba/numba/issues/5282, please follow that ticket for developments.
